### PR TITLE
Disable `String#to_f` special number specs on MSYS2 MINGW64 environment

### PR DESCRIPTION
**Pull request by HertzDevil**
Workaround for https://github.com/crystal-lang/crystal/issues/6170#issuecomment-2450242078.

### CODE REVIEW

File: spec/support/win32.cr [hunk 1]
```diff
@@ -17,3 +17,24 @@ require "spec"
...
+  {% if flag?(:win32) && flag?(:gnu) %}
+    uname = IO::Memory.new
+    if Process.run("uname", %w(-s), output: uname).success?
+      uname = uname.to_s
+      if uname.starts_with?("MINGW64_NT") || uname.starts_with?("MSYS_NT")
+        ENV["MSYSTEM"]?.presence
+      end
+    end
+  {% end %}
+rescue IO::Error
```
Comment (straight-shoota) [inline spec/support/win32.cr:39]: **question:** When would the code in this method raise `IO::Error`?

Comment (HertzDevil) [reply spec/support/win32.cr:39]: when `uname` doesn't exist

Comment (straight-shoota) [reply spec/support/win32.cr:39]: Right. `File::NotFoundError` from `Process.run`. I was focused on `IO::Memory`.

**Review by straight-shoota — APPROVED** [review]

**Comment by HertzDevil** [comment]
Closing in favor of #15195

**Pull request closed without merging**